onnxruntime/core/providers/webnn/builders/impl/lrn_op_builder.cc

```diff
@@ -0,0 +1,149 @@
+
+  NodeAttrHelper helper(node);
+  const float alpha = helper.Get("alpha", 0.0001f);
+  const float beta = helper.Get("beta", 0.75f);
+  const float bias = helper.Get("bias", 1.0f);
+  const uint32_t size = helper.Get("size", 1);
+
+  // Prepare WebNN constants for alpha, beta, bias attributes.
+  emscripten::val alpha_constant = model_builder.CreateOrGetScalarConstant<float>(input_data_type, alpha);
+  emscripten::val beta_constant = model_builder.CreateOrGetScalarConstant<float>(input_data_type, beta);
+  emscripten::val bias_constant = model_builder.CreateOrGetScalarConstant<float>(input_data_type, bias);
+  emscripten::val pow1_constant = model_builder.CreateOrGetScalarConstant<float>(input_data_type, 2);
+
+  /**
+      WebNN doesn't support LRN. So decompose it into a series of ops:
+      X --> Pow --> (Transpose)--> Pad --> AveragePool--> (Transpose) --> Mul --> Add --> Pow --> Div
+             ^           ^                      ^               ^          ^       ^       ^       ^
+             |           |                      |               |          |       |       |       |
+            Y:2      (0,2,3,1)           Kernel:(1,size)     (0,3,1,2)   B:alpha  B:bias B:beta  A:input
+      */
+  //
+  // pow(input, 2)
+  emscripten::val label_options = emscripten::val::object();
+  label_options.set("label", node_name + "_pow1");
+  emscripten::val pow1_output = wnn_builder.call<emscripten::val>("pow", input, pow1_constant, label_options);
+
+  // transpose(pow1_output, permutation=[0, 2, 3, 1])
+  // LRN is one of NHWC layout sentive ops. When preferred layout is NCHW, move dimension 1 to dimension 3 (rightmost).
+  if (model_builder.GetPreferredLayout() == DataLayout::NCHW) {
+    std::vector<uint32_t> perm{0, 2, 3, 1};
+    emscripten::val transpose_options = emscripten::val::object();
+    transpose_options.set("label", node_name + "_transpose_rightmost");
+    transpose_options.set("permutation", emscripten::val::array(perm));
+    pow1_output =
+        wnn_builder.call<emscripten::val>("transpose", pow1_output, transpose_options);
+  }
+
+  // pad(pow1_output, beginning_padding = {0, 0, 0, leading_padding}, ending_padding{0, 0, 0, trailing_padding})
+  // Adding a Pad before averagePool2d and calling AveragePool with pads as 0's.
+  const uint32_t leading_padding = floor((size - 1) / 2);
+  const uint32_t trailing_padding = ceil((size - 1) / 2);
+  std::vector<uint32_t> beginning_padding{0, 0, 0, leading_padding};
+  std::vector<uint32_t> ending_padding{0, 0, 0, trailing_padding};
+  emscripten::val pad_options = emscripten::val::object();
+  pad_options.set("label", node_name + "_pad");
+  emscripten::val pad_output =
+      wnn_builder.call<emscripten::val>("pad", pow1_output, emscripten::val::array(beginning_padding),
+                                        emscripten::val::array(ending_padding), pad_options);
+
+  // averagePool2d(pad_output, pool_potions)
+  const std::vector<uint32_t> kernel_shape{1, size};
```

Comment:
To reduce potential confusion when using initializer lists (because it's unclear whether it's a list of `v = {1, size}` or `v(1, size)`) consider explicitly saying either:

    const std::vector<uint32_t> kernel_shape(1, size);

or

    const std::vector<uint32_t> kernel_shape = {1, size};